### PR TITLE
mapping-saturation: add estimted total to plot

### LINKED ISSUE #628: mapping-saturation: rename "Fitted Data" to something else
The term "Fittede Data" is a bit confusing for the saturation curve. How about something like

- raw OSM data
- modelled saturation curve

also we know what has been done (count, area, legth) so the Y axis should be more descriptive than "value".

Bildschirmfoto vom 2023-06-29 17-18-00

### LINKED ISSUE #629: mapping-saturation: estimated-total line is missing

### LINKED ISSUE #691: mapping-saturation: put legend into figure similar to currentness

### PR DESCRIPTION
**Pull request by Gigaszi**
### Description
Rename "fitted data" in legend to "modelled saturation curve" and plot now shows the estimated total data.

### Corresponding issue
Closes #628 
Closes #629
Closes #691
#688

### New or changed dependencies
- None

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)
- [ ] Improve legend placement
- [ ] Add dashed line to legend

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 - refactor(api): clean-up and rename request models ([#649])
 - refactor(api): move openapi param example to model ([#649])
 - currentness: results with low contributions are now labeled undefined ([#643])
+- -mapping-saturation: now shows the estimated total data in the plot ([#658])
```
Comment (matthiasschaub) [inline CHANGELOG.md:26]:
```suggestion
- mapping-saturation: now shows the estimated total data in the plot ([#658])
```

File: tests/integrationtests/indicators/test_mapping_saturation.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def indicator(self, topic_building_count, feature_germany_heidelberg):
         i.calculate()
         return i
 
-    @pytest.mark.skip(reason="Only for manual testing.")  # comment for manual test
+    # @pytest.mark.skip(reason="Only for manual testing.")  # comment for manual test
```
Comment (matthiasschaub) [inline tests/integrationtests/indicators/test_mapping_saturation.py:114]:
```suggestion
    @pytest.mark.skip(reason="Only for manual testing.")  # comment for manual test
```

File: ohsome_quality_analyst/indicators/mapping_saturation/indicator.py [hunk 3]
```diff
@@ -171,6 +171,24 @@ def create_figure(self) -> None:
             fig.update_yaxes(title_text="Value")
         else:
             fig.update_yaxes(title_text=self.topic.aggregation_type.capitalize())
+
```
Comment (matthiasschaub) [inline ohsome_quality_analyst/indicators/mapping_saturation/indicator.py:174]:
```suggestion

        # plot asymptote
```

File: ohsome_quality_analyst/indicators/mapping_saturation/indicator.py [hunk 4]
```diff
@@ -171,6 +171,24 @@ def create_figure(self) -> None:
...
+            line=dict(color="red", width=2, dash="dash"),
+            name="Estimated total data",
+        )
+        y_max = max(max(self.values), max(self.best_fit.fitted_values))
+        padding_percentage = 0
+        if 0 < 1 - y_max / self.data["best_fit"]["asymptote"] < 0.5:
+            padding_percentage = 1 - y_max / self.data["best_fit"]["asymptote"]
+        padding = y_max * (padding_percentage + 0.05)
+        fig.update_layout(showlegend=True)
+        fig.update_yaxes(range=[min(self.values), y_max + padding])
```
Comment (matthiasschaub) [inline ohsome_quality_analyst/indicators/mapping_saturation/indicator.py:190]:
```suggestion
        asymptote = self.data["best_fit"]["asymptote"]
        fig.add_shape(
            type="line",
            x0=min(self.timestamps),
            x1=max(self.timestamps),
            y0=asymptote,
            y1=asymptote,
            line=dict(color="red", width=2, dash="dash"),
            name="Estimated total data",
        )
        y_max = max(max(self.values), max(self.best_fit.fitted_values))
        padding_percentage = 1 - (y_max / asymptote)
        if 0 < padding_percentage < 0.5:
            padding = y_max * (padding_percentage + 0.05)
        else:
            padding = y_max * 0.05
        fig.update_layout(showlegend=True)
        fig.update_yaxes(range=[min(self.values), y_max + padding])
```

**Review by matthiasschaub — COMMENTED** [review]
Cool,looks good. I added some comments. Also could we add a test case where the asymptote lies way beyond the actual data?

**Review by matthiasschaub — APPROVED** [review]

**Comment by Gigaszi** [comment]
Screenshot from 2023-08-21 16-46-35
Current state of the reworked plot. The plot is always at least 5% higher on the y-axis as the estimated total line. It shoudnt cross the legend in the upper left corner

**Pull request merged**